cmd/abigen/main.go

```diff
@@ -232,8 +238,28 @@ func abigen(c *cli.Context) error {
 			libs[libPattern] = nameParts[len(nameParts)-1]
 		}
 	}
+	// Extract all aliases from the flags
+	if c.GlobalIsSet(aliasFlag.Name) {
+		v := strings.Split(c.GlobalString(aliasFlag.Name), ",")
+		for _, alias := range v {
+			// We support multi-versions for aliasing
+			// e.g.
+			//      foo=bar
+			//      foo:bar
+			index := strings.Index(alias, ":")
+			if index == -1 {
+				index = strings.Index(alias, "=")
+			}
+			// Filter out invalid aliasing
+			if index == -1 || index == 0 || index == len(alias)-1 {
+				continue
+			}
+			aliases[alias[:index]] = alias[index+1:] // Add to the aliasing map
+		}
```

Comment:
Use `(\w+)[:=](\w+)` as a regex here. You can also try to range over the submatches of `(?:(\w+)[:=](\w+))` instead of calling split.

Reply:
Thanks for the awesome suggestion!